### ISSUE TITLE
Error on store submission when building for iOS 14 with library using Xamarin.iOS.SwiftRuntimeSupport package

### ISSUE THREAD

**Issue by uwantfries** [xamarin/XamarinComponents#993]
**Steps**
Create a fresh Xamarin.iOS app with SDK 14.0
Add  a library that has a dependency on Xamarin.iOS.SwiftRuntimeSupport package
Build for a device running iOS 14.0.1
Archive ipa
Distribute ipa to store from archive

**Expected**
Successful store submission

**Observed**
Email after upload:

_Dear Developer,

We identified one or more issues with a recent delivery for your app, "XXXX" YYYY Please correct the following issues, then upload again.

ITMS-90426: Invalid Swift Support - The SwiftSupport folder is missing. Rebuild your app using the current public (GM) version of Xcode and resubmit it.

Best regards,

The App Store Team_

**Comment by angeliglesias7** [comment]
Same problem here. Someone who has fixed it?

**Comment by mattleibow** [comment]
I just had a quick chat with the iOS team and it seems that maybe the VS Mac publisher is not doing something it should. To test, is it possible to publish using Xcode?

**Comment by dalexsoto** [comment]
Hello,

could you try the following?

1. In Visual Studio Form Mac -> Build -> Archive for publishing
2. The archive window will show up in Visual Studio for Mac
3. Open Xcode and in there open the Organizer
4. The Xamarin App archive will show up there so continue the upload steps in Xcode

This is only an issue if you must support iOS < 12.2.

**Comment by angeliglesias7** [comment]
When I archive for publishing, I get the following error:
error MSB6006: "ditto" exited with code 1.
I am going to uninstall Xcode and install again, maybe something went wrong when I installed it.

**Comment by framinosona** [comment]
Could this be ? : 
![image](https://user-images.githubusercontent.com/2520362/107841120-71778380-6db8-11eb-8a0e-1ef7878f6690.png)

Source : https://developer.apple.com/documentation/ios-ipados-release-notes/ios-12_4-release-notes

**Comment by angeliglesias7** [comment]
I solved my problem. I think it could be related with Xcode (12.4). 
After archive for publishing, I had to crete a new folder (SwiftSupport) and then select Products -> Applications, right click, Show Package Contents in app.iOS. Then, go to Frameworks folder. In this folder, there are some libraries.
Then, go to this path:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos
Copy the same libraries and override them in Frameworks folder. Then, copy them again in SwiftSupport folder.
That's it. Then you can distribute your app.
Maybe Xcode generates a wrong version of the libraries, so you have to override them from the correct path (in my case the libraries are stored in swift-5.0, but could be the case that you need to pick them from swift/iphoneos folder).

**Comment by OliverFlecke** [comment]
Thanks @angeliglesias7, this works for me too. 
If it helps anyone else, I made a short script to automate the process. Just replace the `app_name` with yours and provide the script with the path the .xcarchive file after publishing.

```
#!/usr/bin/env sh
xcode_lib_path="/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-5.0/iphoneos"
app_path=$1
app_name=<insert app name>
libs=("$app_path/Products/Applications/$app_name/Frameworks/"*.dylib)

for i in "${libs[@]}"
do
  cp "$xcode_lib_path/$(basename "$i")" "$app_path/SwiftSupport/iphoneos/"
  cp "$xcode_lib_path/$(basename "$i")" "$app_path/Products/Applications/$app_name/Frameworks/"
done
```

**Comment by pmahend1** [comment]
@OliverFlecke do we need to add `Xamarin.Swift` of `Xamarin.iOS.SwiftRuntimeSupport` as well apart from the above script?

**Comment by OliverFlecke** [comment]
Yes, you should include `Xamarin.iOS.SwiftRuntimeSupport` in your project.
I have the package as a transitive dependency in my projects, so I have not tested without it.

**Comment by tompi** [comment]
@angeliglesias7 Hey, having this issue now, "ditto exited with code 1" and Im not sure how you solved this? Seems you got around this somehow and managed to Archive for publishing? Did reinstalling xcode fix it for you?

**Comment by angeliglesias7** [comment]
@tompi Hi. I don't remember how I solved that. In my case reinstalling xcode was not the solution, but your problem has to do with Visual studio, not with Apple. If you want to get more details, you can build using the VS terminal. Run something like this:

msbuild /p:Configuration=AppStore /p:ServerAddress="YOUR MAC ADDRESS" /p:ServerUser="YOUR USER" /p:Platform=iPhone /p:ArchiveOnBuild=true /t:"Build" '.\YOURPROJECTNAME.iOS.csproj'

**Comment by HDnoirod** [comment]
Hi @tompi, I'm having the same issue, "ditto exited with code 1" when I try to archive for publishing. Reinstalling xcode did not work for me either, have you found another solution ?

**Comment by rolfbjarne** [comment]
Adding this to your csproj might fix it:

```xml
<Target Name="_SRSCopySwiftSupport" />
```

Note that you'll also have to make sure the min iOS version (the SupportedOSPlatformVersion property in the csproj) is at least 12.2.
